File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int painted = 0;
        int answer = 0;
        
        for (int wall : section) {
            if (painted < wall) {
                painted = wall + m - 1;
                answer++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.kt
class Solution {
    fun solution(n: Int, m: Int, section: IntArray): Int {
        var painted = 0
        var answer = 0
        
        for (wall in section) {
            if (painted < wall) {
                painted = wall + m - 1
                answer++
            }
        }
        
        return answer
    }
}